pkg/querier/iterators/chunk_merge_iterator.go

```diff
@@ -154,14 +175,12 @@ func (c *chunkMergeIterator) At() (t int64, v float64) {
 	return c.currTime, c.currValue
 }
 
-// Histogram support isn't complete yet
 func (c *chunkMergeIterator) AtHistogram() (int64, *histogram.Histogram) {
-	panic(errors.New("chunkMergeIterator: AtHistogram is not implemented"))
+	return c.currTime, c.currHistogram
```

Comment:
This is breaking our pattern where we panic if the wrong type is requested. Need to add one more field to store the current type , e.g. `currValueType`

(By default we're using the optimized Batch iterator, so adding some more overhead here doesn't matter)